### PR TITLE
2.3 dont prune oplog 1749078

### PR DESCRIPTION
**Pull request by jameinel**
## Description of change

As part of Mongo 3.4 support, it turns out mongo (correctly) now refuses to let us trim the local replicaset oplog while part of a replicaset. However, this broke the tear down which assumed we could delete all databases.
This just updates the testing package which now skips the local oplog collection.

## QA steps

Download Mongo 3.4
```
$ export JUJU_MONGOD=$PATH/TO/3.4/mongod
$ cd featuretests
$ go test --check.v --check.f debugLogDb
```

That test suite was using mongo with `--replSet` because it needs the oplog for how debug-log tracks what changes to display. Without this patch, the test suite fails, but with it, it should pass.

## Documentation changes

None

## Bug reference

https://github.com/juju/testing/pull/135
[lp:1749078](https://bugs.launchpad.net/juju/+bug/1749078)

### CODE REVIEW

File: featuretests/dblog_test.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func (s *debugLogDbSuite) SetUpSuite(c *gc.C) {
 
 func (s *debugLogDbSuite) TearDownSuite(c *gc.C) {
 	// Restart mongod without the replicaset enabled so as not to
-	// affect other test that reply on this mongod instance in this
```
Comment (manadart) [inline featuretests/dblog_test.go]: "... other _tests_ that rely ..."?

Comment (jameinel) [reply featuretests/dblog_test.go]: good catch

**Comment by jameinel** [comment]
$$merge$$

**Comment by wupeka** [comment]
$$merge$$

**Pull request merged**